packages/common/src/services/integrations/jwp/JWPAPIService.ts

```diff
@@ -0,0 +1,130 @@
+import { inject, injectable } from 'inversify';
+
+import StorageService from '../../StorageService';
+
+const INPLAYER_TOKEN_KEY = 'inplayer_token';
+const INPLAYER_IOT_KEY = 'inplayer_iot';
+
+const CONTENT_TYPES = {
+  json: 'application/json',
+  form: 'application/x-www-form-urlencoded',
+};
+
+type RequestOptions = {
+  withAuthentication?: boolean;
+  keepalive?: boolean;
+  contentType?: keyof typeof CONTENT_TYPES;
+  responseType?: 'json' | 'blob';
+  includeFullResponse?: boolean;
+};
+
+@injectable()
+export default class JWPAPIService {
+  private readonly storageService: StorageService;
+
+  private sandbox = true;
+
+  constructor(@inject(StorageService) storageService: StorageService) {
+    this.storageService = storageService;
+  }
+
+  setToken = (token: string, refreshToken = '', expires: number) => {
+    return this.storageService.setItem(INPLAYER_TOKEN_KEY, JSON.stringify({ token, refreshToken, expires }), false);
+  };
+
+  getToken = async () => {
+    const tokenObject = await this.storageService.getItem(INPLAYER_TOKEN_KEY, true, false);
+
+    if (tokenObject) {
+      return tokenObject as { token: string; refreshToken: string; expires: number };
+    }
+
+    return { token: '', refreshToken: '', expires: 0 };
+  };
+
+  removeToken = async () => {
+    await Promise.all([this.storageService.removeItem(INPLAYER_TOKEN_KEY), this.storageService.removeItem(INPLAYER_IOT_KEY)]);
+  };
+
+  isAuthenticated = async () => {
+    const token = await this.getToken();
+
+    return !!token.token && token.expires > Date.now() / 1000;
+  };
+
+  private getBaseUrl = () => (this.sandbox ? 'https://staging-v2.inplayer.com' : 'https://services.inplayer.com');
+
+  private performRequest = async (
+    path: string = '/',
+    method = 'GET',
+    body?: Record<string, unknown>,
+    { contentType = 'form', responseType = 'json', withAuthentication = false, keepalive, includeFullResponse = false }: RequestOptions = {},
+    searchParams?: Record<string, string | number>,
+  ) => {
+    const headers: Record<string, string> = {
+      'Content-Type': CONTENT_TYPES[contentType],
+    };
+
+    if (withAuthentication) {
+      const tokenObject = await this.getToken();
+
+      if (tokenObject.token) {
+        headers.Authorization = `Bearer ${tokenObject.token}`;
+      }
+    }
+
+    const formData = new URLSearchParams();
+
+    if (body) {
+      Object.entries(body).forEach(([key, value]) => {
+        if (value || value === 0) {
+          if (typeof value === 'object') {
+            Object.entries(value as Record<string, string | number>).forEach(([innerKey, innerValue]) => {
+              formData.append(`${key}[${innerKey}]`, innerValue.toString());
```

Comment:
Is it just to support custom register fields?

Reply:
Probably yes. There's a nested `metadata` object that gets updated.